pkg/jobs/registry.go

```diff
@@ -1237,6 +1239,7 @@ func (r *Registry) stepThroughStateMachine(
 				"job %d: resuming with non-nil error", job.ID())
 		}
 		resumeCtx := logtags.AddTag(ctx, "job", job.ID())
+		labels := pprof.Labels("job", fmt.Sprintf("%s id=%d", jobType, job.ID()))
```

Comment:
perhaps use jobType as the label, and jobID as the value?

Reply:
maybe! right now our label keys are constants, see the example below. I'm not saying it has to be constants.. I guess we can do something like:

```
 job_backup: Total 120.0ms
      70.0ms (58.33%): jobid=810503562236035073

 job_auto_span_config_reconciliation:
      10.0ms ( 8.33%): jobid=810469809772462081
```

```
(pprof) tags
 addr: Total 40.0ms
       20.0ms (50.00%): [::1]:49755
       20.0ms (50.00%): internal

 appname: Total 40.0ms
          20.0ms (50.00%): $ cockroach sql
          20.0ms (50.00%): $ internal-job-update

 job: Total 120.0ms
      70.0ms (58.33%): BACKUP id=810503562236035073
      40.0ms (33.33%): BACKUP id=810503570072272897
      10.0ms ( 8.33%): AUTO SPAN CONFIG RECONCILIATION id=810469809772462081

 pebble: Total 10.0ms
         10.0ms (  100%): wal-sync

 range_str: Total 50.0ms
            10.0ms (20.00%): 1/1:/{Min-System/NodeL…}
            10.0ms (20.00%): 10/1:/Table/{6-8}
            10.0ms (20.00%): 32/1:/{NamespaceTab…-Table/32}
            10.0ms (20.00%): 33/1:/Table/3{2-3}
            10.0ms (20.00%): 9/1:/Table/{5-6}

 rangefeed: Total 10.0ms
            10.0ms (  100%): sql-watcher-descriptor-rangefeed

 stmt.no.constants: Total 40.0ms
                    20.0ms (50.00%): BACKUP TO '_'
                    20.0ms (50.00%): UPDATE system.jobs SET payload = $1 WHERE id = $1

 stmt.tag: Total 40.0ms
           20.0ms (50.00%): BACKUP
           20.0ms (50.00%): UPDATE
```

Reply:
I think I like the constant label key.

My first reaction was to say we should have "job: 12345" (just the ID), and maybe a separate "jobType: BACKUP", but looking at the example pprof output, I actually can't imagine when I'd want to know _only_ the ID, so I'm actually 👍  on exactly what we have here. Maybe we _also_ want to have "jobType: BACKUP" so you could aggregate across _all_ backups, but we can add it when we need it; for now it isn't that hard to just add up the one or two backup jobs in my head?

Reply:
> note that the pprof label name is different from the tag - do we care? 

Not really; if anything I'd say the log tag should look like our label, if we wanted to lift the sprintf and use it for both, but I'm not sure it s a big deal.